packages/pluggableWidgets/slider-web/src/utils/useOnChangeDebounced.ts

```diff
@@ -0,0 +1,21 @@
+import { Big } from "big.js";
+import { useCallback, useMemo, useRef } from "react";
+import { debounce, executeAction } from "@mendix/piw-utils-internal";
+import { SliderContainerProps } from "../../typings/SliderProps";
+
+export function useOnChangeDebounced(props: SliderContainerProps) {
+    const valueObjRef = useRef(props.valueAttribute);
+    valueObjRef.current = props.valueAttribute;
+
+    const onChangeEnd = useMemo(() => debounce(() => executeAction(props.onChange), 500), [valueObjRef]);
+
+    const onChange = useCallback(
+        (value: number) => {
+            valueObjRef.current.setValue(new Big(value));
+            onChangeEnd();
+        },
+        [valueObjRef, onChangeEnd]
```

Comment:
Having these refs inside the dependencies seems prone to bugs. It's discouraged in almost all scenarios. Can you double check whether we need them or whether they really do anything?

Reply:
This pattern is just to make onChange have stable ref. But this is not really required in our case... I used this pattern quite a lot and it works pretty good, but if you think this is "discouraged" I'll remove it.

Reply:
- I think the ref will remain stable regardless, because it's not dependent on re-renders or anything.
- This is not a personal opinion, the default React hooks linter also throws warning on including ref objects as hooks dependencies. 😄  As mentioned, if you're sure everything works fine, then I don't mind keeping it in. But as a pattern this is more often than not discouraged AFAIK.
- I do think you want to keep the `useCallback` tho, but I'm not 100% sure on that.